src/testing/testing_suite.nit

```diff
@@ -389,13 +389,13 @@ redef class MMethodDef
 end
 
 redef class MClassDef
-	# Is the class a TestClass?
-	# i.e. is a subclass of `TestSuite`
-	private fun is_test: Bool do
-		var in_hierarchy = self.in_hierarchy
-		if in_hierarchy == null then return false
-		for sup in in_hierarchy.greaters do
-			if sup.name == "TestSuite" then return true
+	# Does self have the `test` annotation?
+	private fun is_test(mbuilder: ModelBuilder): Bool do
+		var anode = mbuilder.mentity2node(self)
+		if not anode isa AClassdef then return false
+		for apropdef in anode.n_propdefs do
+			if not apropdef isa AAnnotPropdef then continue
+			if apropdef.n_atid.n_id.text == "test" then return true
```

Comment:
I'm not a fan since 1. you don't use the annotation facility (`get_annotations` and `process_annotated_node`) and 2. you subordinate the model behavior to the AST
BUT
1. the annotation facility is not optimal and could lead to a more complex code anyway
2. pull is often simpler that push although the usual way would be to make `is_test` a local variable in MClass that would be initialized by a specific processing on the AClassdef